### ISSUE TITLE
Access denied error when accessing C:\vagrant

### ISSUE THREAD

**Issue by EriksonBahr** [hashicorp/vagrant#11186]
### Vagrant version
Vagrant 2.2.6

### Host operating system
Windows 10 pro 1903
Hyper V provider

### Guest operating system
Windows Server Datacenter 2019 1809

### Vagrantfile
```ruby
# -*- mode: ruby -*-
# vi: set ft=ruby :

Vagrant.require_version ">= 1.7.4"
#https://www.howtogeek.com/117192/how-to-run-powershell-commands-on-remote-computers/
Vagrant.configure("2") do |config|
  config.vm.box          = "w2019docker" # pretty much the same as StefanScherer/windows_2019_docker
  config.vm.communicator = "winrm"
  config.vm.synced_folder ".", "/vagrant", type: "smb",  mount_options: ["vers=3"],
  smb_username: "myHostUser", smb_password: "myHostPassword"

  config.vm.provider "hyperv" do |v|
    v.cpus = 2
    v.maxmemory = 2048
    v.linked_clone = true
    v.enable_virtualization_extensions = true
  end

  config.vm.define "registry" do |config|
    config.vm.hostname = "registry"
    config.vm.network :private_network, ip: "172.17.171.50", gateway: "172.17.171.49"
    config.vm.provision "shell", path: "scripts/fix-second-network.ps1", privileged: false, args: "172.17.171.50"
    config.vm.provision "shell", path: "scripts/run-registry.ps1", privileged: false
    config.vm.provision "shell", path: "scripts/run-swarm-master.ps1", privileged: false
    config.vm.provision "shell", path: "scripts/run-portainer.ps1", privileged: false
    config.vm.network :forwarded_port, guest: 3375, host: 3375, id: "swarm", auto_correct: true
    config.vm.network :forwarded_port, guest: 9000, host: 9000, id: "portainer", auto_correct: true
  end

  config.vm.define "sw-win-01" do |config|
    config.vm.hostname = "sw-win-01"
    config.vm.network :private_network, ip: "172.17.171.51", gateway: "172.17.171.49"
    config.vm.provision "shell", path: "scripts/fix-second-network.ps1", privileged: false, args: "172.17.171.51"
    config.vm.provision "shell", path: "scripts/add-registry.ps1", privileged: false
    config.vm.provision "shell", path: "scripts/run-swarm-agent.ps1", privileged: false
  end

  config.vm.define "sw-lin-01" do |config|
    config.vm.hostname     = "sw-lin-01"
    config.vm.box          = "ubuntu/xenial64"
    config.vm.communicator = "ssh"
    config.vm.network :private_network, ip: "172.17.171.52", gateway: "172.17.171.49"
    config.vm.provision "shell", path: "scripts/install-docker.sh", privileged: false
    config.vm.provision "shell", path: "scripts/run-swarm-agent.sh", privileged: false
  end
end

```
### Debug output
https://gist.github.com/EriksonBahr/8193e309b595e6d8798f85c658f8f3b7

### Expected behavior
I should be able to read directories and files inside the smb "C:\vagrant" folder from within the VM.

### Actual behavior
The access denied error is thrown when I try to read the file C:\vagrant\config\swarm-token.

### Steps to reproduce
Vagrant up

### Important notes
I'm able to access the C:\vagrant folder with full access (read and write) via remote desktop. Either via windows explorer and/or command prompt.

That is NOT true when I access via WinRM (either via vagrant or windows EnterPSSession):
vagrant powershell registry
[172.17.171.56]: PS C:\Users\vagrant\Documents> cd C:\vagrant\
[172.17.171.56]: PS C:\vagrant> dir
dir : Access to the path 'C:\vagrant' is denied.
    + CategoryInfo          : PermissionDenied:(C:\vagrant:String) [Get-ChildItem], UnauthorizedAccessException
    + FullyQualifiedErrorId : DirUnauthorizedAccessError,Microsoft.PowerShell.Commands.GetChildItemCommand

### References
None.

**Comment by briancain** [comment]
Does this happen with all Vagrant boxes using winrm, or just that one specifically? @EriksonBahr

**Comment by briancain** [comment]
Also, when you create these shared folders with SMB on your guest, they are likely owned by Administrator. Your local provisioning scripts explicitly are not privileged, which means they probably just don't have permission to access the folder where your synced folder exists. I imagine if you remove that constraint, your provisioners will have access to the folders on your guest.

**Comment by EriksonBahr** [comment]
Hi @briancain 

Thanks for replying so fast... that's the first image I'm building with packer and deploying with vagrant, therefore I did not try others.

You're right! I actually changed the privileged to _true_ and it worked like a charm. Although I had to add a sleep in the first provision script to give some time for the SMB to become available, otherwise I get "Network path not accesssible".

I think the fact I couldn't access this SMB folder via manual-winRM connection via `Enter-PSSession` drove me nuts. Btw... I'm still getting this access denied error, but now I don't care much once I'm able to access via regular means (via provision scripts).

Thanks a lot for the help! Issue closed.
